### PR TITLE
'share' install path added

### PR DESCRIPTION
**Pull request by whit2333**
In an attempt to isolate the source/build/install the directory `CMAKE_INSTALL_PREFIX/share/DD4hep` is used to install all the example data. 

The directory structure in this directory is 
```
└── DD4hep
    ├── DDDetectors
    │   └── compact
    │       └── SiD
    └── examples
        ├── AlignDet
        │   ├── compact
        │   ├── drivers
        │   └── src
        ├── bin
        ├── CLICSiD
        │   ├── compact
        │   ├── eve
        │   ├── scripts
        │   └── sim
        ├── ClientTests
        │   ├── compact
        │   ├── scripts
        │   ├── src
        │   └── src_boost
        ├── Conditions
        │   ├── data
        │   ├── src
        │   └── xml
        ├── DDDB
        │   ├── data
        │   └── scripts
        ├── DDEve
        ├── DDG4
        │   ├── data
        │   └── examples
        ├── lib
        ├── Segmentation
        └── SimpleDetector
            ├── compact
            ├── examples
            └── src
```

There is a bit of overlap since the source directories `DDG4/examples` and `examples/DDG4` are both installing into the `share/DD4hep/examples/DDG4` directory. It seems these examples should be reorganized or removed.

BEGINRELEASENOTES
-  All examples are installed to `CMAKE_INSTALL_PREFIX/share/DD4hep`

ENDRELEASENOTES

### CODE REVIEW

File: cmake/DD4hepConfig.cmake.in [hunk 1]
```diff
@@ -25,9 +25,17 @@ set ( DD4HEP_USE_CXX14   "@DD4HEP_USE_CXX14@" )
 set ( Geant4_DIR         "@Geant4_DIR@" )
 set ( GEANT4_USE_CLHEP   "@GEANT4_USE_CLHEP@" )
 
-set ( ROOTSYS            "$ENV{ROOTSYS}" )
-set ( ROOT_DIR           "$ENV{ROOTSYS}/cmake" )
-set ( ROOT_VERSION       "@ROOT_VERSION@" )
+
+if(NOT ENV{ROOTSYS})
```
Comment (andresailer) [inline cmake/DD4hepConfig.cmake.in]: What does this have to do with the `share` directory? And why set ROOTSYS etc. to DD4hep_DIR?

Comment (whit2333) [inline cmake/DD4hepConfig.cmake.in]: Sorry, yes you are right.  This might not be the most appropriate commit. But here is the reason for doing this. Using `ROOTSYS` (which is really a legacy env variable) in the build will avoid problems on systems where it is not used. For example https://github.com/AIDASoft/DD4hep/issues/162. Ideally only the information in `ROOTConfig.cmake` would be used for finding ROOT dependencies.

Comment (andresailer) [inline cmake/DD4hepConfig.cmake.in]: This also conflicts with #173

**Comment by whit2333** [comment]
See https://github.com/AIDASoft/DD4hep/pull/176
